pandas/_libs/tslibs/offsets.pyx

```diff
@@ -302,6 +302,14 @@ class _BaseOffset(object):
     _normalize_cache = True
     _cacheable = False
     _day_opt = None
+    _attributes = frozenset(['n', 'normalize'])
+
+    @property
+    def kwds(self):
+        # for backwards-compatibility
+        kwds = {name: getattr(self, name, None) for name in self._attributes
```

Comment:
huh?

Reply:
In particular DateOffset has a lot of _attributes that may not get set unless specifically passed.